src/main/java/org/cloudfoundry/promregator/auth/OAuth2XSUAAEnricher.java

```diff
@@ -1,181 +1,103 @@
 package org.cloudfoundry.promregator.auth;
 
 import java.io.IOException;
-import java.io.UnsupportedEncodingException;
-import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
-import java.util.ArrayList;
-import java.util.Base64;
-import java.util.List;
-
-import org.apache.http.Consts;
-import org.apache.http.NameValuePair;
-import org.apache.http.ParseException;
-import org.apache.http.client.ClientProtocolException;
+
 import org.apache.http.client.config.RequestConfig;
-import org.apache.http.client.entity.UrlEncodedFormEntity;
-import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.methods.HttpPost;
-import org.apache.http.impl.client.CloseableHttpClient;
-import org.apache.http.impl.client.HttpClients;
-import org.apache.http.message.BasicNameValuePair;
-import org.apache.http.util.EntityUtils;
 import org.cloudfoundry.promregator.config.OAuth2XSUAAAuthenticationConfiguration;
-
-import com.google.gson.Gson;
-import com.google.json.JsonSanitizer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class OAuth2XSUAAEnricher implements AuthenticationEnricher {
-	
+
 	private static final Logger log = LoggerFactory.getLogger(OAuth2XSUAAEnricher.class);
-	
-	static final CloseableHttpClient httpclient = HttpClients.createDefault();
 
-	private final OAuth2XSUAAAuthenticationConfiguration config;
-	
+	private final TokenFetcher tokenFetcher;
+
 	public OAuth2XSUAAEnricher(OAuth2XSUAAAuthenticationConfiguration config) {
 		super();
-		this.config = config;
+		this.tokenFetcher = setupTokenFetcher(config);
+	}
+
+	private final static TokenFetcher setupTokenFetcher(OAuth2XSUAAAuthenticationConfiguration config) {
```

Comment:
Minor: Unclear to me why you are going for a "private-scoped factory pattern" here.

Reply:
Setup of token client is now done inside contructor itself.